lib/impure/db_sqlite.nim

```diff
@@ -713,11 +720,70 @@ proc setEncoding*(connection: DbConn, encoding: string): bool {.
   exec(connection, sql"PRAGMA encoding = ?", [encoding])
   result = connection.getValue(sql"PRAGMA encoding") == encoding
 
-proc finalize*(sqlPrepared:SqlPrepared){.discardable.} = 
+proc finalize*(sqlPrepared:SqlPrepared) {.discardable, since: (1, 3).} = 
   discard finalize(sqlPrepared.PStmt)
 
+template dbBindParamError*(paramIdx: int, val: untyped) =
+  ## Raises a `DbError` exception.
+  var e: ref DbError
+  new(e)
+  e.msg = "error binding param in position " & $paramIdx 
+  raise e
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int32) {.since: (1, 3).} =
+  ## Binds a int32  to the specified paramIndex.
+  if bind_int(ps.PStmt, paramIdx.int32, val) != SQLITE_OK:
+    dbBindParamError(paramIdx, val)
+
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int) {.since: (1, 3).} =
+  ## Binds a int  to the specified paramIndex.
+  when sizeof(int) == 8:
+    if bind_int64(ps.PStmt, paramIdx.int32, val.int64) != SQLITE_OK:
+      dbBindParamError(paramIdx, val)
+  else:
+    if bind_int(ps.PStmt, paramIdx.int32, val.int32) != SQLITE_OK:
+      dbBindParamError(paramIdx, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: int64) {.since: (1, 3).} =
+  ## Binds a int64  to the specified paramIndex.
+  if bind_int64(ps.PStmt, paramIdx.int32, val) != SQLITE_OK:
+    dbBindParamError(paramIdx, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: float64) {.since: (1, 3).} =
+  ## Binds a 64bit float to the specified paramIndex.
+  if bind_double(ps.PStmt, paramIdx.int32, val) != SQLITE_OK:
+    dbBindParamError(paramIdx, val)
+
+proc bindNull*(ps: SqlPrepared, paramIdx: int) {.since: (1, 3).} =
+  ## Sets the bindparam at the specified paramIndex to null 
+  ## (default behaviour by sqlite).
+  if bind_null(ps.PStmt, paramIdx.int32) != SQLITE_OK:
+    dbBindParamError(paramIdx, val)
+
+proc bindParam*(ps: SqlPrepared, paramIdx: int, val: string) {.since: (1, 3).} =
+  ## Binds a string to the specified paramIndex.
+  if bind_text(ps.PStmt, paramIdx.int32, val.cstring, -1.int32, SQLITE_STATIC) != SQLITE_OK:
```

Comment:
Our strings may be NUL-terminated, but that doesn't stop anyone from adding NUL to their strings. Instead of making SQLite guess the length, pass them the length that we already know.

```suggestion
  if bind_text(ps.PStmt, paramIdx.int32, val.cstring, val.len.int32, SQLITE_STATIC) != SQLITE_OK:
```

I still think this should be `bind_blob` however.

Since we are using `SQLITE_STATIC` here, a warning to potential users would be nice.

Alternatively, maybe introduce a `copy` parameter at the end which dictates whether the data should be copied by SQLite? We can give it a default value of `true` to be on the safe side.